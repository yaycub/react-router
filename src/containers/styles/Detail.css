.Detail{
  margin: 0 auto;
  margin-top: 10vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 18px;
  border: 10px dotted black;
  border-radius: 5px;
  width: 25vw;
  padding: 10px;

  figure{
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
  }

  img{
    border: 5px double black;
    padding: 20px;
    margin: 5px;
  }

  ul{
    text-align: center;

    li{
      margin: 5px;
    }
  }
}