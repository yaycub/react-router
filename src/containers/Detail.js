import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/Detail.css';
import { useOneCharacter } from '../hooks/useOneCharacter';

const Detail = ({ match }) => {
  const oneCharacter = useOneCharacter(match.params.id);
  const { name, status, species, origin, location, image } = oneCharacter;
  
  return (
    <article className={styles.Detail}>
      <figure>
        <img src={image} alt={name} />
        <figcaption>{name}</figcaption>
      </figure>
      <ul>
        <li>Statue: {status}</li>
        <li>Species: {species}</li>
        <li>Origin: {origin}</li>
        <li>Location: {location}</li>
      </ul>
    </article>
  );
};

Detail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Detail;
