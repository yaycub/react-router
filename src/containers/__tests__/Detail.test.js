import React from 'react';
import { shallow } from 'enzyme';
import Detail from '../Detail';

jest.mock('../../services/getCharacters');

describe('Detail Component', () => {
  it('should render Detail', () => {
    const wrapper = shallow(<Detail match={{ params: { id: '1' } }} />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
