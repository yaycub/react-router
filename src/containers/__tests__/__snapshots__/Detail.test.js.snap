// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Component should render Detail 1`] = `
<article>
  <figure>
    <img />
    <figcaption />
  </figure>
  <ul>
    <li>
      Statue: 
    </li>
    <li>
      Species: 
    </li>
    <li>
      Origin: 
    </li>
    <li>
      Location: 
    </li>
  </ul>
</article>
`;
